File: Android/AnyMind/app/src/main/java/com/anymind/anymind/sync/SyncModels.kt
package com.anymind.anymind.sync

import com.google.gson.annotations.SerializedName


data class SyncChange(
    @SerializedName("id") val id: String,
    @SerializedName("content") val content: String,
    @SerializedName("system_tags") val systemTags: List<String>,
    @SerializedName("user_tags") val userTags: List<String>,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String,
    @SerializedName("deleted") val deleted: Boolean,
    @SerializedName("base_rev") val baseRev: Long?
)

data class SyncPushRequest(
    @SerializedName("space_id") val spaceId: String,
    @SerializedName("space_secret") val spaceSecret: String,
    @SerializedName("device_id") val deviceId: String,
    @SerializedName("changes") val changes: List<SyncChange>
)

data class SyncPushResult(
    @SerializedName("id") val id: String,
    @SerializedName("server_rev") val serverRev: Long,
    @SerializedName("server_updated_at") val serverUpdatedAt: String,
    @SerializedName("conflict") val conflict: Boolean
)

data class SyncPushResponse(
    @SerializedName("results") val results: List<SyncPushResult>,
    @SerializedName("server_rev_max") val serverRevMax: Long
)

data class SyncPullRequest(
    @SerializedName("space_id") val spaceId: String,
    @SerializedName("space_secret") val spaceSecret: String,
    @SerializedName("since_rev") val sinceRev: Long,
    @SerializedName("limit") val limit: Int
)

data class SyncPullChange(
    @SerializedName("id") val id: String,
    @SerializedName("content") val content: String,
    @SerializedName("system_tags") val systemTags: List<String>,
    @SerializedName("user_tags") val userTags: List<String>,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String,
    @SerializedName("deleted") val deleted: Boolean,
    @SerializedName("server_rev") val serverRev: Long,
    @SerializedName("server_updated_at") val serverUpdatedAt: String
)

data class SyncPullResponse(
    @SerializedName("changes") val changes: List<SyncPullChange>,
    @SerializedName("server_rev_max") val serverRevMax: Long
)

data class SpaceCreateRequest(
    @SerializedName("name") val name: String?
)

data class SpaceCreateResponse(
    @SerializedName("space_id") val spaceId: String,
    @SerializedName("space_secret") val spaceSecret: String,
    @SerializedName("created_at") val createdAt: String
)

File: ios/AnyMind/PromptRecorder/Sync/SyncModels.swift
import Foundation

struct SyncChange: Codable {
    let id: String
    let content: String
    let systemTags: [String]
    let userTags: [String]
    let createdAt: String
    let updatedAt: String
    let deleted: Bool
    let baseRev: Int64?

    enum CodingKeys: String, CodingKey {
        case id
        case content
        case systemTags = "system_tags"
        case userTags = "user_tags"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case deleted
        case baseRev = "base_rev"
    }
}

struct SyncPushRequest: Codable {
    let spaceId: String
    let spaceSecret: String
    let deviceId: String
    let changes: [SyncChange]

    enum CodingKeys: String, CodingKey {
        case spaceId = "space_id"
        case spaceSecret = "space_secret"
        case deviceId = "device_id"
        case changes
    }
}

struct SyncPushResult: Codable {
    let id: String
    let serverRev: Int64
    let serverUpdatedAt: String
    let conflict: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case serverRev = "server_rev"
        case serverUpdatedAt = "server_updated_at"
        case conflict
    }
}

struct SyncPushResponse: Codable {
    let results: [SyncPushResult]
    let serverRevMax: Int64

    enum CodingKeys: String, CodingKey {
        case results
        case serverRevMax = "server_rev_max"
    }
}

struct SyncPullRequest: Codable {
    let spaceId: String
    let spaceSecret: String
    let sinceRev: Int64
    let limit: Int

    enum CodingKeys: String, CodingKey {
        case spaceId = "space_id"
        case spaceSecret = "space_secret"
        case sinceRev = "since_rev"
        case limit
    }
}

struct SyncPullChange: Codable {
    let id: String
    let content: String
    let systemTags: [String]
    let userTags: [String]
    let createdAt: String
    let updatedAt: String
    let deleted: Bool
    let serverRev: Int64
    let serverUpdatedAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case content
        case systemTags = "system_tags"
        case userTags = "user_tags"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case deleted
        case serverRev = "server_rev"
        case serverUpdatedAt = "server_updated_at"
    }
}

struct SyncPullResponse: Codable {
    let changes: [SyncPullChange]
    let serverRevMax: Int64

    enum CodingKeys: String, CodingKey {
        case changes
        case serverRevMax = "server_rev_max"
    }
}

struct SpaceCreateRequest: Codable {
    let name: String?
}

struct SpaceCreateResponse: Codable {
    let spaceId: String
    let spaceSecret: String
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case spaceId = "space_id"
        case spaceSecret = "space_secret"
        case createdAt = "created_at"
    }
}
